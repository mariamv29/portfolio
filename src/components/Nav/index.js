import React from "react";

function Nav() {
    const tabs = ['Home', 'About', 'Work', 'Contact']
  return (
    <header className="flex-row px-1">
      <h2>
        <a data-testid="link" href="/">
          <span arial-label="home">Maria Mireles</span>
        </a>
      </h2>
      <nav>
        <ul className="flex-row">
          <li className="mx-2">
            <a data-testid="about" href="#about">About me</a>
          </li>
          <li>
          <a data-testid="mywork" href="#mywork">Portfolio</a>
          </li>
          <li>
          <a data-testid="contact" href="#contact">Contact</a>
          </li>
        </ul>
      </nav>
    </header>
  );
}

export default Nav;
