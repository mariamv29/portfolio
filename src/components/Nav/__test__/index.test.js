import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Nav from "..";

afterEach(cleanup);

describe("Nav component renders", () => {
  it("renders", () => {
    render(<Nav />);
  });
  it("matches snapshot", () => {
    const { asFragment } = render(<Nav />);
    expect(asFragment()).toMatchSnapshot();
  });
});
describe("links are visible", () => {
  it("inserts text into the links", () => {
    const { getByTestId } = render(<Nav />);
    expect(getByTestId("link")).toHaveTextContent("Maria Mireles");
    expect(getByTestId("about")).toHaveTextContent("About me");
    expect(getByTestId("mywork")).toHaveTextContent("Portfolio");
    expect(getByTestId("contact")).toHaveTextContent("Contact");
  });
});
