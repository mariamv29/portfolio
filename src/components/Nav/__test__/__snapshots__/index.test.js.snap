// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component renders matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="flex-row px-1"
  >
    <h2>
      <a
        data-testid="link"
        href="/"
      >
        <span
          arial-label="home"
        >
          Maria Mireles
        </span>
      </a>
    </h2>
    <nav>
      <ul
        class="flex-row"
      >
        <li
          class="mx-2"
        >
          <a
            data-testid="about"
            href="#about"
          >
            About me
          </a>
        </li>
        <li>
          <a
            data-testid="mywork"
            href="#mywork"
          >
            Portfolio
          </a>
        </li>
        <li>
          <a
            data-testid="contact"
            href="#contact"
          >
            Contact
          </a>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
